src/test/serialize_tests.cpp

```diff
@@ -376,4 +377,175 @@ BOOST_AUTO_TEST_CASE(class_methods)
     BOOST_CHECK(methodtest3 == methodtest4);
 }
 
+enum class BaseFormat {
+    DEC,
+    HEX,
+};
+
+/// (Un)serialize a number as a string either from/to decimal or hexadecimal.
+class Base
+{
+public:
+    uint8_t m_base_data;
+
+    Base() : m_base_data(17) {}
+    explicit Base(uint8_t data) : m_base_data(data) {}
+
+#if 0 // the following two are equivalent
+    SERIALIZE_METHODS_PARAMS(Base, obj, BaseFormat, fmt)
+    {
+        if (ser_action.ForRead()) {
+            std::string str;
+            s >> str;
+            assert(str.size() == 2 || (fmt == BaseFormat::DEC && str.size() == 3));
+            uint32_t data;
+            bool ok;
+            if (fmt == BaseFormat::DEC) {
+                ok = ParseUint32(str, &data);
```

Comment:
In commit "test: add tests that exercise WithParams()" (483ed457411d164ad71115dbfe1d2b4ff8b2e5f3)

Think this needs to capitalize Int: `ParseUInt32`